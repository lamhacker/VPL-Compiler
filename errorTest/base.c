#include "base.h"

float* mallocVec(int num)
{
    float* vec = malloc(sizeof(float) * num + SSE_ALIGN);
	vec = (float *) align(vec, SSE_ALIGN);
	return vec;
}

void freeVec(float* vec)
{
    free(vec);
}

void setValue(float* vec, float val, int num)
{
    for (int i = 0; i < num; i++)
    {
        vec[i] = val;
    }
}

void printVec(char* rep, float* vec, int num)
{
	printf("%s: ", rep);
    for (int i = 0; i < num; i++)
    {   
        printf("%f ", vec[i]);
    }
	printf("\n");
}


