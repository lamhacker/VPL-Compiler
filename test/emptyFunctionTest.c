#include "base.h"
#define NUM (4)

extern void empty(long, float*);

int main(void) {
	
	float* a = mallocVec(NUM);
	
	printf("function: \n");
    setValue(a, 0, NUM);
    empty(NUM, a);
	printVec("a", a, NUM);
	
	freeVec(a);
	
	return 0;
}
