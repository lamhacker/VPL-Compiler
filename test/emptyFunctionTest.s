
.text
.global empty
.type empty, @function
.p2align 4,,15

empty:
    pushq   %rbp
    movq    %rsp, %rbp
    pushq   %rbx

    # allocate 0 local variables
    movq %rdi, %rax
    imulq $4, %rax, %rax
    addq $16, %rax
    imulq $0, %rax, %rax
    subq %rax, %rsp
    andq $-16, %rsp

    

    popq    %rbx
    leave
    ret

