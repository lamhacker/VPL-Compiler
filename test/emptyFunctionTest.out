function: 
a: 0.000000 0.000000 0.000000 0.000000 
